pylint/extensions/kwargs.py

```diff
@@ -0,0 +1,93 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from astroid import BoundMethod, UnboundMethod, nodes
+
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import safe_infer
+from pylint.interfaces import INFERENCE
+
+if TYPE_CHECKING:
+    from pylint.lint import PyLinter
+
+
+class KeywordChecker(BaseChecker):
+    """Checks related to keyword arguments."""
+
+    name = "kwargs"
+    msgs = {
+        "W3501": (
+            "Call to `%s` missing keyword argument `%s`.",
+            "consider-using-keyword-argument",
+            "When using a literal directly in a function call, it's very hard to know what the argument is"
+            " if a positional argument is used because there's no variable name or attribute name to "
+            "rely on when reading. Using a keyword argument there's at least the attribute name to "
+            "help the reader understanding the call.",
+        ),
+    }
+
+    def visit_call(self, node: nodes.Call) -> None:
+        """Check that called functions/methods are provided keyword arguments."""
+        called = safe_infer(node.func)
+
+        if not called:
+            return
+
+        needed_keywords, default_kwarg_names = self._get_args(called)
+
+        if len(needed_keywords) - len(default_kwarg_names) < 2:
+            # This checker will not apply if there are less than 2 positional args.
```

Comment:
This feels quite arbitrary. Why was this limit chosen?

Reply:
@Pierre-Sassoulas 's suggestion in this PR and in issue.

Reply:
Shall we make it configurable?

Reply:
For example if we take ``join`` or ``enumerate`` or ``range`` everyone knows about the first positional argument, it's the most used one and it's the obvious one. If we're going to raise for those kind of call, then the false positive rate will be enormous. As soon as there is more than one argument I think the call can benefit from keyword args for litterals. Right now it's hard to be sure of what we need to do because the primers are spammed by call without any args/kwargs with litteral in the default value. I think we should rethink this once we can see the actual primer result.

Reply:
The primer result is out and it looks pretty reasonable to me, what do you think?

Reply:
Not sure if I'm looking at the wrong primer but we should not raise here https://github.com/PyCQA/astroid/blob/b717e99964bf9e601b0acb939865990e64da19ee/astroid/bases.py#L160. There's no positional literal argument used in this call. If we can't differentiate genuine explicit positional value from implicit default values the check will not be usable.